src/mca/rmaps/round_robin/rmaps_rr_mappers.c

```diff
@@ -441,11 +447,14 @@ int prte_rmaps_rr_bycpu(prte_job_t *jdata, prte_app_context_t *app,
             rc = prte_rmaps_base_check_oversubscribed(jdata, app, node, options);
             if (PRTE_ERR_TAKE_NEXT_OPTION == rc) {
                 /* move to next node */
+                PMIX_RELEASE(proc);
                 break;
             } else if (PRTE_SUCCESS != rc) {
                 /* got an error */
+                PMIX_RELEASE(proc);
                 goto errout;
             }
+            PMIX_RELEASE(proc);
```

Comment:
Yes proc has a reference count of 2 when returning from prte_rmaps_base_setup_proc, one when allocating the object and the other when adding the proc to the node proc array.

Reply:
Correct - because it gets released once when we cleanup the node proc array, and again when we cleanup the job proc array. So why do we need this extra release? Is it because we aren't properly cleaning up one of the objects that contains this one in its array?

Reply:
Actually, I see what is going on - the proc object actually exits the "setup_proc" function with a reference count of 3. One when it get allocated, the second when it gets added to the node procs array, and a third right before returning from the function. The proc object doesn't actually get added to its associated job object's proc array.

Let me take a gander for a bit - I don't immediately recall why I did it that way. Could be that the problem you are trying to solve is that the proc object isn't getting correctly inserted into the job proc array.

Reply:
I think the PMIX_RETAIN before returning from the function is for the node procs array since it wasn't retained just before adding it. I think proc has a reference count 2 after calling the function. Yes,  the proc object doesn't actually get added to its associated job object's proc array. That's why there is a PMIX_RETAIN on line 355 in file [src/mca/rmaps/seq/rmaps_seq.c].

Reply:
Think I might have an alternative path - let me explore it a bit more, post a draft PR for you to look at for comparison.